### PR TITLE
Add more logging in the replicating connectors

### PR DESCRIPTION
**Pull request by tbak**
### Description of the Change

Additional logging in the stream replication connectors, that should help diagnose
reoccurring stability issues.

### CODE REVIEW

File: titus-server-runtime/src/main/java/com/netflix/titus/runtime/connector/agent/replicator/GrpcAgentReplicatorEventStream.java [hunk 1]
```diff
@@ -56,7 +59,10 @@ public GrpcAgentReplicatorEventStream(AgentManagementClient client,
                     metrics.event(titusRuntime.getClock().wallTime() - event.getLastUpdateTime());
                 })
                 .doOnUnsubscribe(metrics::disconnected)
-                .doOnError(metrics::disconnected)
+                .doOnError(error -> {
+                    logger.warn("Connection to the agent event stream terminated with an error: {}", error.getMessage(), error);
```
Comment (fabiokung) [inline titus-server-runtime/src/main/java/com/netflix/titus/runtime/connector/agent/replicator/GrpcAgentReplicatorEventStream.java]: nit: `logger.error`?

File: titus-server-runtime/src/main/java/com/netflix/titus/runtime/connector/eviction/replicator/GrpcEvictionReplicatorEventStream.java [hunk 2]
```diff
@@ -53,7 +53,10 @@ public GrpcEvictionReplicatorEventStream(EvictionServiceClient client,
     @Override
     public Observable<ReplicatorEvent<EvictionDataSnapshot>> connect() {
```
Comment (fabiokung) [inline titus-server-runtime/src/main/java/com/netflix/titus/runtime/connector/eviction/replicator/GrpcEvictionReplicatorEventStream.java]: all these `connect` methods seem very similar. Consider factoring them out?

File: titus-server-runtime/src/main/java/com/netflix/titus/runtime/connector/agent/replicator/GrpcAgentReplicatorEventStream.java (hunk 1 above)
Comment (tbak) [reply titus-server-runtime/src/main/java/com/netflix/titus/runtime/connector/agent/replicator/GrpcAgentReplicatorEventStream.java]: Connection failure is most of the time non-critical issue (could be master failover, gateway termination, etc).

File: titus-server-runtime/src/main/java/com/netflix/titus/runtime/connector/eviction/replicator/GrpcEvictionReplicatorEventStream.java (hunk 2 above)
Comment (tbak) [reply titus-server-runtime/src/main/java/com/netflix/titus/runtime/connector/eviction/replicator/GrpcEvictionReplicatorEventStream.java]: +1
I think it is the right time to do it.

**Pull request merged**